clang/lib/CIR/Dialect/Transforms/TargetLowering/Targets/AArch64.cpp

```diff
@@ -62,6 +62,21 @@ class AArch64TargetLoweringInfo : public TargetLoweringInfo {
       : TargetLoweringInfo(std::make_unique<AArch64ABIInfo>(LT, Kind)) {
     assert(!MissingFeature::swift());
   }
+
+  unsigned getTargetAddrSpaceFromCIRAddrSpace(
+      mlir::cir::AddressSpaceAttr addressSpaceAttr) const override {
+    using Kind = mlir::cir::AddressSpaceAttr::Kind;
+    switch (addressSpaceAttr.getValue()) {
+    case Kind::offload_private:
+    case Kind::offload_local:
+    case Kind::offload_global:
+    case Kind::offload_constant:
+    case Kind::offload_generic:
+      return 0;
+    default:
+      llvm_unreachable("Unknown CIR address space for this target");
```

Comment:
Don't we want to fail unconditionally?

Reply:
Although it's a bit weird, Clang allows compile OpenCL to general targets like `x86_64` and `aarch64`. In [that case](https://godbolt.org/z/378ozGqME), all address spaces are mapped to `addrspace(0)`.

I don't think such results make any sense for production use, but let's align to it anyway😂 Wdyt?

Reply:
Ok, yeah makes sense to align with the existing behavior then.

Reply:
Yes, for example `clang/test/CodeGenOpenCL/address-spaces-conversions.cl`:
```
// RUN: %clang_cc1 %s -triple x86_64-unknown-linux-gnu -O0 -ffake-address-space-map -cl-std=CL2.0 -emit-llvm -o - | FileCheck %s
// RUN: %clang_cc1 %s -triple x86_64-unknown-linux-gnu -O0 -ffake-address-space-map -cl-std=CL3.0 -cl-ext=+__opencl_c_generic_address_space -emit-llvm -o - | FileCheck %s
// RUN: %clang_cc1 %s -triple x86_64-unknown-linux-gnu -O0 -cl-std=CL2.0 -emit-llvm -o - | FileCheck --check-prefix=CHECK-NOFAKE %s
// RUN: %clang_cc1 %s -triple x86_64-unknown-linux-gnu -O0 -cl-std=CL3.0 -cl-ext=+__opencl_c_generic_address_space -emit-llvm -o - | FileCheck --check-prefix=CHECK-NOFAKE %s
// When -ffake-address-space-map is not used, all addr space mapped to 0 for x86_64.
```

> What happens for this target if the user passes an attribute address space with a random number?

Similar to the behavior in this PR, it will be reflected in the LLVM IR regardless of the target.